src/main/scala/sbtdocker/models.scala

```diff
@@ -97,10 +103,15 @@ case class ImageName(
   override def toString = {
     val registryString = registry.fold("")(_ + "/")
     val namespaceString = namespace.fold("")(_ + "/")
-    val tagString = tag.fold("")(":" + _)
+    val tagString = sanitizedTagString.fold("")(":" + _)
     registryString + namespaceString + repository + tagString
   }
 
+  private def sanitizedTagString = tag.map(s => {
```

Comment:
I suggest to do this sanitization in the apply method in the companion object if it should be transparent. Then there is a way to skip it, by calling `new ImageName(...)`.